benchmarks/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -436,13 +439,29 @@ int main_(std::vector<std::string> const &args, const MPI_Comm comm)
 
   if (perform_radius_search)
   {
-    // radius chosen in order to control the number of results per query
-    // NOTE: minus "1+sqrt(3)/2 \approx 1.37" matches the size of the boxes
-    // inserted into the tree (mid-point between half-edge and
-    // half-diagonal)
-    double const r =
-        2. * std::cbrt(static_cast<double>(n_neighbors) * 3. / (4. * M_PI)) -
-        (1. + std::sqrt(3.)) / 2.;
+    // Radius is computed so that the number of results per query for a
+    // uniformly distributed points in a [-a,a]^d box is approximately
+    // n_neighbors. This requires adjusting as construction is based on
+    // [-1,1]^d boxes. Instead of an exact calculation using integrals,
+    // an approximation is made by averaging half-edge and half-diagonal).
+    double r = 0.;
+    switch (partition_dim)
+    {
+    case 1:
+      // n_values*(2*r)/(2a) = n_neighbors
+      r = static_cast<double>(n_neighbors) - 1.;
+      break;
+    case 2:
+      // n_values*(M_PI*r^2)/(2a)^2 = n_neighbors
+      r = std::sqrt(static_cast<double>(n_neighbors) * 4. / M_PI) -
+          (1. + std::sqrt(2.)) / 2;
+      break;
+    case 3:
+      // n_values*(4/3*M_PI*r^3)/(2a)^3 = n_neighbors
+      r = std::cbrt(static_cast<double>(n_neighbors) * 6. / M_PI) -
+          (1. + std::cbrt(3.)) / 2;
```

Comment:
Can you please comment a little more on the last term in all three cases?

Reply:
I'm not sure what you want me to add to the comment above. This is the average of half-edge and half-diagonal.

Reply:
I would prefer the formula to match what we implement.
So instead of just taking `r` from it, we make a little bit smaller and I am still a little bit confused about why this is necessary.

Reply:
The main thing why this is tricky is because in this driver we deal with boxes and not points. So, it is not sufficient to just consider the density of box centers, as even the boxes with centers that are farther apart may overlap. Thus, we need to decrease the radius.

The formula there is just for the density of of point cloud, i.e. first term. I found it helpful to understand that term. I can make an explicit comment about that.

Reply:
Yes, please add something explaining that the primitives here are boxes and not points. Keeping that in mind it makes sense to have such a kind of correction. (On the other hand, I would not be opposed to just use points instead of boxes but that's for a later discussion I think).

Reply:
The comment above states: `This requires adjusting as construction is based on [-1,1]^d boxes.` I'm not sure what else I can add. I guess I can put another sentence there.

Reply:
Something like
```
    // Radius is computed so that the number of results per query for a
    // uniformly distributed points in a [-a,a]^d box is approximately
    // n_neighbors. 
    // Note that the primitives considered here are not points but boxes.
    // The radius is decreased by averaging the length of a half-edge 
    // and the length of the half-diagonal to account for that approximately.
```
would work for me.